libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/DefaultOnMissedCallNotificationHandler.kt

```diff
@@ -30,8 +30,9 @@ class DefaultOnMissedCallNotificationHandler @Inject constructor(
         // Resolve the event and add a notification for it, at this point it should no longer be a ringing one
         val notificationData = matrixClientProvider.getOrRestore(sessionId).getOrNull()
             ?.notificationService()
-            ?.getNotification(roomId, eventId)
+            ?.getNotifications(sessionId, mapOf(roomId to listOf(eventId)))
```

Comment:
It looks weird to have to provide a `sessionId` here, since we are using a method from a service of a matrix client.

Reply:
True, but `RustNotificationService` doesn't have a reference to the matrix client. Maybe I should just add a `SessionId` as a constructor param for it. WDYT?

Reply:
> Maybe I should just add a `SessionId` as a constructor param for it. WDYT?

Yes please. We're doing it for `RustRoomFactory` already for instance.

Reply:
`RustRoomFactory` doesn't use injection though. Adding it here would mean creating a factory abstraction.

Reply:
https://github.com/element-hq/element-x-android/pull/4722/commits/b5ece84b52dc9730af6f5a1041e9717761766c39